src/cmd/run.go

```diff
@@ -63,13 +67,61 @@ var runCmd = &cobra.Command{
 			message.Fatalf(err, "Cannot unmarshal %s", config.TaskFileLocation)
 		}
 
-		if config.ListTasks {
+		if config.ListTasks || config.ListAllTasks {
 			rows := [][]string{
 				{"Name", "Description"},
 			}
 			for _, task := range tasksFile.Tasks {
 				rows = append(rows, []string{task.Name, task.Description})
 			}
+			// If ListAllTasks, add tasks from included files
+			if config.ListAllTasks {
+				var includedTasksFile types.TasksFile
+				var fullPath string
+				templatePattern := `\${[^}]+}`
+				re := regexp.MustCompile(templatePattern)
+				for _, include := range tasksFile.Includes {
+					// get included TasksFile
+					for includeName, includeFileLocation := range include {
+						// check for templated variables in includeFileLocation value
+						if re.MatchString(includeFileLocation) {
+							runner := runner.Runner{
+								TemplateMap: map[string]*zarfUtils.TextTemplate{},
+								TasksFile:   tasksFile,
+								TaskNameMap: map[string]bool{},
+							}
+							runner.PopulateTemplateMap(runner.TasksFile.Variables, config.SetRunnerVariables)
+							includeFileLocation = runner.TemplateString(includeFileLocation)
+						}
+						// check if included file is a url
+						if helpers.IsURL(includeFileLocation) {
+							// If file is a url download it to a tmp directory
+							tmpDir, err := zarfUtils.MakeTempDir(zarfConfig.CommonOptions.TempDirectory)
+							defer os.RemoveAll(tmpDir)
+							if err != nil {
+								message.Fatalf(err, "error removing %s", tmpDir)
+							}
+							fullPath = filepath.Join(tmpDir, filepath.Base(includeFileLocation))
```

Comment:
Question about using `filepath.Base`. What happens it we have 2 remote `includes` both of which end in `tasks.yaml`? Would one clobber the other?

Reply:
Will add a test for this to confirm